src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareUtils.cs

```diff
@@ -0,0 +1,151 @@
+            diffEntry.UpdateAction = difference.UpdateAction;
+            diffEntry.DifferenceType = difference.DifferenceType;
+            diffEntry.Name = difference.Name;
+
+            if (difference.SourceObject != null)
+            {
+                diffEntry.SourceValue = GetName(difference.SourceObject.Name.ToString());
+                var sourceType = new SchemaComparisonExcludedObjectId(difference.SourceObject.ObjectType, difference.SourceObject.Name);
+                diffEntry.SourceObjectType = sourceType.TypeName;
+
+            }
+            if (difference.TargetObject != null)
+            {
+                diffEntry.TargetValue = GetName(difference.TargetObject.Name.ToString());
+                var targetType = new SchemaComparisonExcludedObjectId(difference.TargetObject.ObjectType, difference.TargetObject.Name);
+                diffEntry.TargetObjectType = targetType.TypeName;
+            }
+
+            if (difference.DifferenceType == SchemaDifferenceType.Object)
+            {
+                // set source and target scripts
+                if (difference.SourceObject != null)
+                {
+                    string sourceScript;
+                    difference.SourceObject.TryGetScript(out sourceScript);
+                    diffEntry.SourceScript = FormatScript(sourceScript);
+                }
+                if (difference.TargetObject != null)
+                {
+                    string targetScript;
+                    difference.TargetObject.TryGetScript(out targetScript);
+                    diffEntry.TargetScript = FormatScript(targetScript);
+                }
+            }
+
+            diffEntry.Children = new List<DiffEntry>();
+
+            foreach (SchemaDifference child in difference.Children)
+            {
+                diffEntry.Children.Add(CreateDiffEntry(child, diffEntry));
+            }
+
+            return diffEntry;
+        }
+
+        internal static SchemaCompareEndpoint CreateSchemaCompareEndpoint(SchemaCompareEndpointInfo endpointInfo, string connectionString)
+        {
+            switch (endpointInfo.EndpointType)
+            {
+                case SchemaCompareEndpointType.Dacpac:
+                    {
+                        return new SchemaCompareDacpacEndpoint(endpointInfo.PackageFilePath);
+                    }
+                case SchemaCompareEndpointType.Database:
+                    {
+                        return new SchemaCompareDatabaseEndpoint(connectionString);
+                    }
+                default:
+                    {
+                        return null;
+                    }
+            }
+        }
+
+        internal static string GetConnectionString(ConnectionInfo connInfo, string databaseName)
+        {
+            if (connInfo == null)
+            {
+                return null;
+            }
+
+            connInfo.ConnectionDetails.DatabaseName = databaseName;
```

Comment:
not new code but as I understand copy is done because connInfo created from ownerURI doesnt have db information... and i think it was not done in caller here just to avoid duplicating this line...